drivers/bluetooth/hci/rpmsg.c

```diff
@@ -78,14 +78,16 @@ static struct net_buf *bt_rpmsg_evt_recv(const uint8_t *data, size_t remaining)
 	}
 	BT_DBG("len %u", hdr.len);
 
-	buf = bt_buf_get_evt(hdr.evt, discardable, K_NO_WAIT);
+	buf = bt_buf_get_evt(hdr.evt, discardable,
+			     discardable ? K_NO_WAIT : K_FOREVER);
+
```

Comment:
I can't really return anything meaningful here: the caller can't return anything since it's an IPC callback (workqueue context), and returning without doing anything means the buffer is dropped. So I will loop through with 10 seconds as you suggest, having a warning on each timeout.